enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/ShiroAuthManager.java

```diff
@@ -160,14 +161,10 @@ public AuthSubject login( String username, String password )
             try
             {
                 subject.login( token );
-            }
-            catch ( ExpiredCredentialsException e )
-            {
-                result = AuthenticationResult.PASSWORD_CHANGE_REQUIRED;
-
-                // We have to build an identity with the given username to allow the user to change password
-                // At this point we know that the username is valid
-                subject = buildSubject( username );
+                if ( realm.findUser( username ).passwordChangeRequired() )
+                {
+                    result = AuthenticationResult.PASSWORD_CHANGE_REQUIRED;
+                }
```

Comment:
This moves us slightly away from multi-realm authentication, since it is assuming the login was handled by FileUserRealm